content/docs/2.7/scalers/gcp-storage.md

```diff
@@ -0,0 +1,148 @@
+You can use `TriggerAuthentication` CRD to configure the authenticate by providing the service account credentials in JSON. 
+
+**Credential based authentication:**
+
+- `GoogleApplicationCredentials` - Service account credentials in JSON.
+
+### Example
+
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: gcp-storage-scaledobject
+  namespace: keda-gcp-storage-test
+spec:
+  scaleTargetRef:
+    name: keda-gcp-storage-go
+  triggers:
+  - type: gcp-storage
+    metadata:
+      bucketName: "Transactions"
+      targetObjectCount: "5"
+      credentialsFromEnv: GOOGLE_APPLICATION_CREDENTIALS_JSON
+```
+
+### Example using TriggerAuthentication
+
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: keda-trigger-auth-gcp-credentials
+spec:
+  secretTargetRef:
+  - parameter: GoogleApplicationCredentials 
+    name: gcp-storage-secret        # Required. Refers to the name of the secret
+    key: GOOGLE_APPLICATION_CREDENTIALS_JSON       # Required.
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: gcp-storage-scaledobject
+spec:
+  scaleTargetRef:
+    name: keda-gcp-storage-go
+  triggers:
+  - type: gcp-storage
+    authenticationRef:
+      name: keda-trigger-auth-gcp-credentials
+    metadata:
+      bucketName: "Transactions"
+      targetObjectCount: "5"
+```
+
+**Identity based authentication:**
+
+You can also use `TriggerAuthentication` CRD to configure the authentication using the associated service account of the running machine in Google Cloud. You only need to create a `TriggerAuthentication` as this example, and reference it in the `ScaledObject`. `ClusterTriggerAuthentication` can also be used if you intend to use it globally in your cluster.
+
+### Example using TriggerAuthentication with GCP Identity
+
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: keda-trigger-auth-gcp-credentials
+spec:
+  podIdentity:
+    provider: gcp
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: gcp-storage-scaledobject
+spec:
+  scaleTargetRef:
+    name: keda-gcp-storage-go
+  triggers:
+  - type: gcp-storage
+    authenticationRef:
+      name: keda-trigger-auth-gcp-credentials
+    metadata:
+      bucketName: "Transactions"
+      targetObjectCount: "5"
+```
+
+## Example using ClusterTriggerAuthentication with GCP Identity
+
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: ClusterTriggerAuthentication
+metadata:
+  name: keda-clustertrigger-auth-gcp-credentials
+spec:
+  podIdentity:
+    provider: gcp
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: gcp-storage-scaledobject
+spec:
+  scaleTargetRef:
+    name: keda-gcp-storage-go
+  triggers:
+  - type: gcp-storage
+    authenticationRef:
+      name: keda-clustertrigger-auth-gcp-credentials
+      kind: ClusterTriggerAuthentication
+    metadata:
+      bucketName: "Transactions"
+      targetObjectCount: "5"
+```
```

Comment:
If we cover TriggerAuthentication the nI think we already provide enough help, but can't hurt